packages/node_modules/@webex/internal-plugin-services/src/interceptors/service.js

```diff
@@ -0,0 +1,102 @@
+/*!
+ * Copyright (c) 2015-2019 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {Interceptor} from '@webex/http-core';
+
+const pattern = /(?:^\/)|(?:\/$)/;
+
+/**
+ * @class
+ */
+export default class ServiceInterceptor extends Interceptor {
+  /**
+   * @returns {ServiceInterceptor}
+   */
+  static create() {
+    /* eslint no-invalid-this: [0] */
+    return new ServiceInterceptor({webex: this});
+  }
+
+  /* eslint-disable no-param-reassign */
+  /**
+   * @see Interceptor#onRequest
+   * @param {Object} options
+   * @returns {Object}
+   */
+  onRequest(options) {
+    const {services} = this.webex.internal;
+    let service;
+
+    // Populate option object's tag key.
+    // A later PR will handle generating these tags at
+    // request construction. These tags are used to validate
+    // if a request has already been handled by a specific
+    // interceptor. This is useful when multiple interceptors
+    // operate on the same options object keys.
+    if (!options.tags) { options.tags = {}; }
+
+    // If an operation hasn't been made via
+    // this interceptor...
+    if (!options.tags.serviceInterceptor) {
+      options.tags.serviceInterceptor = true;
+
+      // Some plugins are sending requests directly via the
+      // `uri` key. To not conflict with the `internal-plugin-wdm`
+      // interceptors, we need to check the following two key values
+      // to confirm an operation needs to be made.
+      if (!options.service && !options.resource) {
+        return options;
+      }
+
+      // Check if the `services` plugin is currently ready.
+      // If not, wait until it is and continue.
+      services.waitUntilReady()
+        .then(() => {
+          this.validateOptions(options);
+
+          service = services.get(options.service);
+
+          if (!service) {
+            throw new Error(`\`${options.service}\` is not a known service`);
+          }
+
+          options.uri = this.generateUrl(service, options.resource);
+
+          return options;
+        });
+    }
+
+    return options;
+  }
+  /* eslint-enable no-param-reassign */
+
+  /* eslint-disable class-methods-use-this */
+  /**
+   * Verify that all required parameters have been specified.
+   * @param {object} options
```

Comment:
Could you add the params the options object is expected to have as new jsdoc lines like:
```
* @param {String} options.resource
```